### PR TITLE
Add version and changelog for SecureDrop 2.7.0~rc1

### PR DESCRIPTION
**Pull request by zenmonkeykstop**
## Status

Ready for review 

## Description of Changes

Towards #6979.

- Syncs dev and prod markupsafe dependency
- Adds 2.7.0 changelog

## Testing

- [ ] version incremented correctly
- [ ] changelog looks good.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -2,6 +2,43 @@
 
 ## 2.7.0~rc1
 
+### Web applications
+* Use Sequioa and redwood instead of gnupg and pretty_bad_protocol for GPG operations (#6891, #6884, #6913, #6912, #6925, #6926, #6949, #6958, #6892, #6948, #6946, #6970, #6975, #6972, #6983, #6981, #6998)
```
Comment (rocodes) [inline changelog.md]: nit: Sequoia

File: changelog.md [hunk 2]
```diff
@@ -2,6 +2,43 @@
...
+### Development
+
+* Dependencies changes:
+  * Remove boto and boto3 dependencies (#6890)
+  * Remove hypothesis dependency (#6893)
+  * Update certifi from 2022.12.7 to 2023.7.22 (#6900)
+  * Update pillow from 9.3.0 to 10.0.1 (#6959)
+* Improve printing of apparmor denials in testinfra suite (#6883)
+* Set "ia" as unavailable locale, "eo" as test locale  (#6919)
+* Adde script to auto-backport PRs to release branches (#6875)
```
Comment (rocodes) [inline changelog.md]: Nit: Add

File: changelog.md (hunk 1 above)
Comment (zenmonkeykstop) [reply changelog.md]: Squished!

File: changelog.md (hunk 2 above)
Comment (zenmonkeykstop) [reply changelog.md]: Squished!

**Review by rocodes — APPROVED** [review]
LGTM. Looks like CI failure also happening on main so fix will need to be backported.

**Pull request merged**